pyspeckit/spectrum/units.py

```diff
@@ -319,14 +319,17 @@ def __new__(self, xarr, unit='Hz', refX=None, redshift=None,
         subarr._unit = self.validate_unit(unit, bad_unit_response)
         subarr.refX = refX
 
+        if hasattr(refX, 'unit'):
+            if refX_unit != refX.unit:
+                raise ValueError("If refX is a Quantity then refX_unit must be None or the same as refX.unit")
+            refX_unit = refX.unit 
         if refX_unit is None:
             if subarr._unit in frequency_dict:
                 refX_unit = subarr.unit
             else:
                 refX_unit = 'Hz'
-            subarr.refX_unit = refX_unit
-        else:
-            subarr.refX_unit = refX_unit
+        subarr.refX_unit = refX_unit
```

Comment:
I think this is backwards - there should not be a `subarr.refX_unit`, instead there should be only a `subarr.refX` that has a `subarr.refX.unit`.